src/renderer/components/+workspaces/workspace-list.tsx

```diff
@@ -0,0 +1,90 @@
+import "./workspaces.scss";
+
+import React from "react";
+import { RouteComponentProps } from "react-router";
+import { observer } from "mobx-react";
+import { navigation } from "../../navigation";
+import { workspaceListURL, IWorkspaceListRouteParams } from "./workspace-list.route";
+import { TabLayout } from "../layout/tab-layout";
+import { Badge } from "../badge";
+import { ItemListLayout, ItemListLayoutProps } from "../item-object-list/item-list-layout";
+import { AddWorkspaceDialog } from "./add-workspace-dialog";
+import { Workspace, WorkspaceId } from "../../../common/workspace-store";
+import { WorkspaceItem, workspaceListStore } from "./workspace-list.store";
+import { WorkspaceDetails } from "./workspace-details";
+
+enum sortBy {
+  name = "name",
+  description = "description",
+}
+
+interface Props extends RouteComponentProps<IWorkspaceListRouteParams> {
+}
+
+@observer
+export class WorkspaceList extends React.Component<Props> {
+  componentDidMount() {
+    workspaceListStore.loadAll();
+  }
+
+  get selectedWorkspace() {
+    const { match: { params: { workspaceName } } } = this.props;
+
+    return workspaceListStore.getByName(workspaceName);
+  }
+
+  showDetails = (workspace: WorkspaceItem) => {
+    if (!workspace) {
+      navigation.merge(workspaceListURL());
+    }
+    else {
+      navigation.merge(workspaceListURL({
+        params: {
+          workspaceName: workspace.getName(),
+        }
+      }));
+    }
+  };
+
+  hideDetails = () => {
+    this.showDetails(null);
+  };
+
+  render() {
+    return (
+      <TabLayout>
+        <ItemListLayout
```

Comment:
@aleksfront or @ixrock, any idea why this element only fills about a third of the width available (see video)? @Nokel81 and I couldn't figure it out...